### PR TITLE
update speech synthesis format

### PR DESCRIPTION
**Pull request by yulin-li**
- Add `Raw24Khz16BitMonoTrueSilk`, mark siren format as unsupported
- Call `onAudioEnd()` callback if the format is not supported for playback
- Update riff header for mulaw

### CODE REVIEW

File: src/sdk/Audio/SpeakerAudioDestination.ts [hunk 1]
```diff
@@ -85,27 +87,40 @@ export class SpeakerAudioDestination implements IAudioDestination, IPlayer {
...
-                const audioBlob = new Blob([receivedAudio], { type: AudioFormatToMimeType[this.privFormat.formatTag] });
-                this.privAudio.src = window.URL.createObjectURL(audioBlob);
-                this.notifyPlayback().then(() => {
-                    if (!!cb) {
-                        cb();
+            } else {
+                let receivedAudio = new ArrayBuffer(this.privBytesReceived);
+                this.privAudioOutputStream.read(receivedAudio).then((_: number): void => {
+                    if (this.privFormat.hasHeader) {
+                        receivedAudio = SynthesisAdapterBase.addHeader(receivedAudio, this.privFormat);
```
Comment (glharper) [inline src/sdk/Audio/SpeakerAudioDestination.ts:100]: Doesn't addHeader already perform the hasHeader() check?

Comment (yulin-li) [reply src/sdk/Audio/SpeakerAudioDestination.ts:100]: The `receivedAudio` is from `privAudioOutputStream` and has no header. So if the format needs a header, we need to add it here and send the audio chunk with header to browser for playback.

Comment (glharper) [reply src/sdk/Audio/SpeakerAudioDestination.ts:100]: My comment is about the line:
 `
if (this.privFormat.hasHeader) {`

The first three lines of SynthesisAdapterBase.addHeader are:
```
    public static addHeader(audio: ArrayBuffer, format: AudioOutputFormatImpl): ArrayBuffer {
        if (!format.hasHeader) {
            return audio;
        }
```
Do we need to perform the hasHeader check twice?

Comment (yulin-li) [reply src/sdk/Audio/SpeakerAudioDestination.ts:100]: Yes, you are right, I removed the duplicate checking. Thanks!

**Review by glharper — APPROVED** [review]

**Pull request merged**